relayer/publish.go

```diff
@@ -0,0 +1,147 @@
+package relayer
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	legacyerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	txtypes "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/rollchains/tiablob/celestia/appconsts"
+	blobtypes "github.com/rollchains/tiablob/celestia/blob/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+const (
+	CelestiaPublishKeyName  = "blob"
+	CelestiaFeegrantKeyName = "feegrant"
+	celestiaBech32Prefix    = "celestia"
+	celestiaBlobPostMemo    = "Posted by tiablob https://rollchains.com"
+)
+
+// postNextBlocks will post the next block (or chunk of blocks) above the last proven height to Celestia (does not include the block being proposed).
+// Skip the last n blocks to give time for the previous proposer's transaction to succeed.
+func (r *Relayer) postNextBlocks(ctx sdk.Context, n int) {
+	// TODO post blocks since last proven height
+	height := ctx.BlockHeight()
+
+	if height <= 1 {
+		return
+	}
+
+	// only publish every n blocks
+	if (height-1)%int64(n) != 0 {
+		return
+	}
+
+	if !r.provider.KeyExists(CelestiaPublishKeyName) {
```

Comment:
Can we do this check earlier, e.g. in NewRelayer()?

Reply:
we want to check every time so that the operator can manage their keys while the process is running.